<?xml version="1.0"?>

<launch>
  <arg name="ns" default="vsemi_c320_camera"/>
  <arg name="nn" default="camera"/>
  <arg name="respawn" default="false"/>  
  <arg name="lens_type" default="0"/>
  <arg name="lens_center_offset_x" default="0"/>
  <arg name="lens_center_offset_y" default="0"/>
  <arg name="image_type" default="1"/>
  <arg name="start_stream" default="true"/>
  <arg name="hdr_mode"  default="0"/>
  <arg name="int0"  default="1200"/>
  <arg name="int1"  default="200"/>
  <arg name="int_gray" default="0"/>
  <arg name="offset" default="0"/>
  <arg name="min_amplitude" default="60"/>
  <arg name="frequency_modulation" default="2"/>
  <arg name="channel" default="0"/>
  <arg name="median_filter" default="false"/>
  <arg name="average_filter" default="false"/>
  <arg name="edge_filter_threshold" default="0"/>
  <arg name="temporal_filter_factor" default="0.0"/>
  <arg name="temporal_filter_threshold" default="0"/>
  <arg name="temporal_edge_filter_threshold_low" default="0"/>
  <arg name="temporal_edge_filter_threshold_high" default="0"/>
  <arg name="interference_detect_limit" default="0"/>
  <arg name="use_last_value" default="false"/>
  <arg name="cartesian" default="true"/>
  <arg name="publish_point_cloud" default="true"/>
  <arg name="video_camera" default="/dev/v4l/by-id/usb-046d_Logitech_Webcam_C930e_986A01BE-video-index0"/>
  <arg name="imu_port" default="/dev/ttyIMU"/>

  <group ns="$(arg ns)">
    <node pkg="vsemi_c320_camera"
          type="c320_camera_node"
          name="$(arg nn)"
          output="screen"
          respawn="$(arg respawn)">

      <param name="lens_type" value="$(arg lens_type)"/>
      <param name="lens_center_offset_x" value="$(arg lens_center_offset_x)"/>
      <param name="lens_center_offset_y" value="$(arg lens_center_offset_y)"/>
      <param name="image_type" value="$(arg image_type)"/>
      <param name="start_stream" value="$(arg start_stream)"/>
      <param name="hdr_mode" value="$(arg hdr_mode)"/>
      <param name="int0" value="$(arg int0)"/>
      <param name="int1" value="$(arg int1)"/>
      <param name="int_gray" value="$(arg int_gray)"/>
      <param name="offset" value="$(arg offset)"/>
      <param name="min_amplitude" value="$(arg min_amplitude)"/>
      <param name="frequency_modulation" value="$(arg frequency_modulation)"/>
      <param name="channel" value="$(arg channel)"/>
      <param name="median_filter" value="$(arg median_filter)"/>
      <param name="average_filter" value="$(arg average_filter)"/>
      <param name="edge_filter_threshold" value="$(arg edge_filter_threshold)"/>
      <param name="temporal_filter_factor" value="$(arg temporal_filter_factor)"/>
      <param name="temporal_filter_threshold" value="$(arg temporal_filter_threshold)"/>
      <param name="temporal_edge_filter_threshold_low" value="$(arg temporal_edge_filter_threshold_low)"/>
      <param name="temporal_edge_filter_threshold_high" value="$(arg temporal_edge_filter_threshold_high)"/>
      <param name="interference_detect_limit" value="$(arg interference_detect_limit)"/>
      <param name="use_last_value" value="$(arg use_last_value)"/>
      <param name="cartesian" value="$(arg cartesian)"/>
      <param name="publish_point_cloud" value="$(arg publish_point_cloud)"/>
      <param name="video_camera" value="$(arg video_camera)"/>
      <param name="imu_port" value="$(arg imu_port)"/>

    </node>

    <node name="rviz" pkg="rviz" type="rviz" args= "-d $(find vsemi_c320_camera)/rviz/vsemi_c320_camera.rviz" > </node>
    <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure"> </node>
    <node name="base_link" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 map sensor_frame 1" />

  </group>
</launch>
